### PR TITLE
[VM] Fix the shape function of conv & Add dynamic support for conv2d nhwc

### PR DESCRIPTION
**Pull request by jcf94**
Just a small bug fix when using vm with conv2d_nhwc. Also add the shape function for `nn.fast_softmax.`

cc @comaniac

### CODE REVIEW

File: python/tvm/relay/op/nn/_nn.py [hunk 1]
```diff
@@ -975,23 +976,72 @@ def _conv_shape_func(dshape, kshape, strides, padding, dilation):
...
+                    inputs[0],
+                    inputs[1],
+                    convert(strides),
+                    convert(padding),
+                    convert(dilation),
+                )
+            ]
+    raise ValueError(
+        "Unsupported data/kernel layout: %s, %s" % (attrs["data_layout"], attrs["kernel_layout"])
+    )
```
Comment (comaniac) [inline python/tvm/relay/op/nn/_nn.py:1044]:
```suggestion
    shape_func = None
    if attrs["data_layout"] == "NCHW" and attrs["kernel_layout"] == "OIHW":
        shape_func = conv_shape_func_nchw
    elif attrs["data_layout"] == "NHWC" and attrs["kernel_layout"] == "HWIO":
        shape_func = conv_shape_func_nhwc_hwio
    elif attrs["data_layout"] == "NHWC" and attrs["kernel_layout"] == "HWOI":
        shape_func = conv_shape_func_nhwc_hwoi
    else:
        raise ValueError(
            "Unsupported data/kernel layout: %s, %s" % (attrs["data_layout"], attrs["kernel_layout"])
        )
        
    return [shape_func(inputs[0], inputs[1], convert(strides), convert(padding), convert(dilation))]
```

File: python/tvm/topi/arm_cpu/conv2d_spatial_pack.py [hunk 2]
```diff
@@ -273,7 +273,8 @@ def conv2d_spatial_pack_nhwc(cfg, data, kernel, strides, padding, dilation, out_
     data_pad = nn.pad(data, [0, pad_top, pad_left, 0], [0, pad_down, pad_right, 0])
 
     # ==================== define configuration space ====================
-    n, oc, oh, ow = cfg.axis(N), cfg.axis(OC), cfg.axis(OH), cfg.axis(OW)
+    n = cfg.axis(N) if isinstance(N, int) else cfg.axis(1)
```
Comment (comaniac) [inline python/tvm/topi/arm_cpu/conv2d_spatial_pack.py:276]: Does it mean to use `1` as the batch size when it is dynamic? Better to add a comment if so.

**Review by comaniac — APPROVED** [review]
LGTM. Just some style suggestions.

File: python/tvm/topi/arm_cpu/conv2d_spatial_pack.py (hunk 2 above)
Comment (jcf94) [reply python/tvm/topi/arm_cpu/conv2d_spatial_pack.py:277]: Comment added. This is to set the split factor candidate to 1 when the batch size is dynamic.

**Pull request merged**